### ISSUE TITLE
for "jsonschema lint", what causes "unexpected error: map::at"?

### ISSUE THREAD

**Issue by kussmaul** [sourcemeta/jsonschema#360]
I'm new to `jsonschema lint` and like it - thank you! However, I sometimes get `"unexpected error: map::at"` which appears to comes from `main.cc` but is hard to debug, and `lint --verbose` seems to have no effect. As shown below, I tried to find the offending file without success. (The `"line 1 and column 1"` errors below are for `/**` comments in config files.) Suggestions?

            $ npx jsonschema lint -i .angular -i node_modules
            unexpected error: map::at
            Please report it at https://github.com/sourcemeta/jsonschema
            
            $ ls tsconfig*.json
            tsconfig.app.json  tsconfig.doc.json  tsconfig.json  tsconfig.spec.json
            
            $ npx jsonschema lint tsconfig*.json
            error: Failed to parse the JSON document at line 1 and column 1
              tsconfig.app.json
            
            $ npx jsonschema lint tsconfig*.json -i tsconfig.app.json
            unexpected error: map::at
            Please report it at https://github.com/sourcemeta/jsonschema
            
            $ npx jsonschema lint tsconfig.app.json tsconfig.doc.json tsconfig.json tsconfig.spec.json
            error: Failed to parse the JSON document at line 1 and column 1
              tsconfig.app.json
            
            $ npx jsonschema lint tsconfig.doc.json tsconfig.json tsconfig.spec.json
            error: Failed to parse the JSON document at line 1 and column 1
              tsconfig.doc.json
            
            $ npx jsonschema lint tsconfig.json tsconfig.spec.json
            error: Failed to parse the JSON document at line 1 and column 1
              tsconfig.json
            
            $ npx jsonschema lint tsconfig.spec.json
            error: Failed to parse the JSON document at line 1 and column 1
              tsconfig.spec.json

            $ npx jsonschema lint tsconfig*.json -i tsconfig.app.json -i tsconfig.doc.json -i tsconfig.json -i tsconfig.spec.json
            unexpected error: map::at
            Please report it at https://github.com/sourcemeta/jsonschema

**Comment by jviotti** [comment]
Hi @kussmaul , thanks for reporting this. Looks pretty weird. Are you running this in an open-source project I can try myself by any chance?

> (The "line 1 and column 1" errors below are for /** comments in config files.)

This makes sense, as JSON does not support comments in the official grammar. There are some JSON parsers that opt-in to be overly permissive, but its not in theory spec compliant!

**Comment by jviotti** [comment]
Oh, I think it has do to with the ignore option in its short variant. See this, which I'm running on the JSON Schema CLI repo itself:

```sh
$ jsonschema lint --ignore action.yml
error: Could not determine the base dialect of the schema

$ jsonschema lint -i action.yml
unexpected error: map::at:  key not found
Please report it at https://github.com/sourcemeta/jsonschema
```

Probably a silly thing, and by looking at this, I bet we are only running the longer option form in the tests 🤦🏻

**Comment by jviotti** [comment]
@kussmaul Please try v9.3.6: https://github.com/sourcemeta/jsonschema/releases/tag/v9.3.6, and let me know if you ever face any other issues. Thanks again!

**Comment by kussmaul** [comment]
v9.3.6 fixes the problem. Thank you for the very quick response!

**Comment by kussmaul** [comment]
For future reference: Since `jsonschema` stops on the first error and doesn't always show filenames, 
it can help to run one file at a time. The (long) `find` command below:
- prunes 2 unwanted paths
- prints the name of each `.json`, `.jsonc`, or `.json5` file
- runs `npx jsonschema lint` on each file with local schema files
 
`find . \( -path ./.angular -o -path ./node_modules \) -prune -o -type f -regex '.*\.json[5c]?' -print -exec npx jsonschema lint {} --resolve src/app/data-model/schema \;`
